beetsplug/lastgenre/__init__.py

```diff
@@ -397,12 +419,16 @@ def _get_genre(
 
         # Return with a combined or freshly fetched genre list.
         if new_genres:
-            suffix = "whitelist" if self.whitelist else "any"
-            label += f", {suffix}"
-
-            if keep_genres:
-                label = f"keep + {label}"
-            return self._combine_genres(keep_genres, new_genres), label
+            resolved_genres = self._combine_resolve_and_log(
+                keep_genres, new_genres
+            )
+            if any(resolved_genres):
+                suffix = "whitelist" if self.whitelist else "any"
+                label += f", {suffix}"
+                if keep_genres:
+                    label = f"keep + {label}"
+                if g_string := self._format_and_stringify(resolved_genres):
+                    return g_string, label
```

Comment:
I think this conditional may be redundant, since the condition above already ensured that `resolved_genres` contains non-empty elements.

```suggestion
                return self._format_and_stringify(resolved_genres), label
```

Reply:
yes you are absolutely right, I was being paranoid. I already removed it again.